crates/m3/src/builder/witness.rs

```diff
@@ -394,8 +396,13 @@ enum WitnessColumnInfo<T> {
 type WitnessDataMut<'a, P> = WitnessColumnInfo<&'a mut [P]>;
 
 impl<'a, P: PackedField> WitnessDataMut<'a, P> {
-	pub fn new_owned(allocator: &'a bumpalo::Bump, log_underlier_count: usize) -> Self {
-		Self::Owned(allocator.alloc_slice_fill_default(1 << log_underlier_count))
+	pub fn new_owned(allocator: &'a HostBumpAllocator<'a, P>, log_underlier_count: usize) -> Self {
+		let slice = allocator
+			.alloc(1 << log_underlier_count)
+			.expect("failed to allocate witness data slice");
+		fill_zeroes(slice);
```

Comment:
Is this strictly necessary?

There is a significant slow down coming from `alloc_slice_fill_default` → `alloc_zeroed` → `memset(0)` in my profiles.

Reply:
Yes, this call is redundant here